### PR TITLE
2844 image chooser block default title

### PR DESCRIPTION
**Pull request by allcaps**
Issue: https://github.com/wagtail/wagtail/issues/2844

Before submitting, please review the contributor guidelines <http://docs.wagtail.io/en/latest/contributing/index.html> and check the following:
Done

* Do the tests still pass? (http://docs.wagtail.io/en/latest/contributing/developing.html#testing)
Ran 3000 tests in 177.495s
OK (skipped=199, expected failures=8)
Destroying test database for alias 'default'...

* Does the code comply with the style guide? (Run `make lint` from the Wagtail root)
Done

* For Python changes: Have you added tests to cover the new/fixed behaviour?
No change

* For front-end changes: Did you test on all of Wagtail’s supported browsers? 
I tested:
- OSX Chrome 64
- OSX Firefox 58.0.2
- OSX Safari 11.0.3
- Win7 IE11
- Win10 IE11
- Win10 Edge

* For new features: Has the documentation been updated accordingly?
Not needed.

### CODE REVIEW

File: wagtail/images/templates/wagtailimages/chooser/chooser.js [hunk 1]
```diff
@@ -109,6 +109,24 @@ function(modal) {
...
+            if (title === '') {
+                // The file widget value example: `C:\fakepath\image.jpg`
+                var parts = fileWidget.val().split('\\');
+                var fileName = parts[parts.length - 1];
+                titleWidget.val(fileName);
+            }
+        });
+    }
+
+    popuplateTitle(modal.body);
```
Comment (zerolab) [inline wagtail/images/templates/wagtailimages/chooser/chooser.js]: `popuplateTitle` ➜ `populateTitle`

File: wagtail/images/templates/wagtailimages/chooser/chooser.js [hunk 2]
```diff
@@ -109,6 +109,24 @@ function(modal) {
...
+    function popuplateTitle(context) {
+        // Note: There are two inputs with `#id_title` on the page.
+        // The page title and image title. Select the input inside the model body.
+        var fileWidget = $('#id_file', context);
+        fileWidget.on('change', function () {
+            var titleWidget = $('#id_title', context);
+            var title = titleWidget.val();
+            if (title === '') {
+                // The file widget value example: `C:\fakepath\image.jpg`
+                var parts = fileWidget.val().split('\\');
```
Comment (BertrandBordage) [inline wagtail/images/templates/wagtailimages/chooser/chooser.js]: Will this handle UNIX paths as well? It doesn’t seem so.

Comment (allcaps) [reply wagtail/images/templates/wagtailimages/chooser/chooser.js]: @BertrandBordage I'm on unix. OSX is certified unix right? It is a weird path. What value does http://jsfiddle.net/QPUJa/ give you when you select a file? If it is something else maybe I can create a better js.

Comment (BertrandBordage) [reply wagtail/images/templates/wagtailimages/chooser/chooser.js]: Yeah, OSX is based on UNIX standards.
I also get a weird `C:\fakepath\filename`. I’m not sure whether it’s coming from jQuery or directly JavaScript, but that’s clearly misleading. You don’t have to do anything about it, then :)

Comment (gasman) [reply wagtail/images/templates/wagtailimages/chooser/chooser.js]: `C:\fakepath` is actually a quirk required by the HTML specification: https://html.spec.whatwg.org/multipage/input.html#file-upload-state-(type=file)

Comment (allcaps) [reply wagtail/images/templates/wagtailimages/chooser/chooser.js]: Do I need to do something to get this merged? Just let me know ;)

Comment (BertrandBordage) [reply wagtail/images/templates/wagtailimages/chooser/chooser.js]: About the path, do nothing, but about `populateTitle` above, there is still one commit missing :)

Comment (allcaps) [reply wagtail/images/templates/wagtailimages/chooser/chooser.js]: Mmm, I didn't get the Github UI, I thought zerolab fixed the typo. :D

**Comment by BertrandBordage** [comment]
Shall we backport this in 2.0.1?

**Comment by thibaudcolas** [comment]
Merged! Thank you @allcaps, welcome to the CONTRIBUTORS list 😄. This will make it into Wagtail 2.1, which isn't that far from now so I don't think it's worth backporting.

A note about future contributions: please don't include the release notes / CONTRIBUTORS / CHANGELOG changes in the PR. It makes the PR much harder to review.

**Pull request merged**